project_management/estimating.md

```diff
@@ -0,0 +1,65 @@
+# Estimating
+
+Estimates are key to what we do.
+
+We are in the business of selling software creation hours. We want to do this as inexpensively as possible while maintaining a healthy revenue for growth and downtimes. As a group, we need to hold appropriate tension between cost and quality.
+
+Estimates should be made by two developers at a minimum; estimates are shared, and ideally by the implementing team.
+When you win and lose together, your estimates get even better.
+
+## Derive duration
+
+We focus on size estimates (how much), not duration estimates (how long). Related, but useful independently.
+Estimates are a combination of effort, risk, complexity, etc. Separating size from duration enables agility in scheduling and iteration planning.
+
+## Split the work
+
+Before the work can be estimated, it has to be split into chunks of a manageable size. Depending on how reliable it needs to be (the estimate type), we will create user stories encompassing a larger amount of functionality or more detailed user stories that define a specific functionality. The team must focus on providing just enough details so that the client can make the right decision.
+
+## T-shirt sizes
+
+T-shirt sizes is the method we use to estimate software size.
+
+The estimating team:
+
+1. **Take one user story at a time.**
+2. **Ensure that everyone involved knows what the end result will be like once the story is completed.** To achieve this, the team will discuss the feature until they have just enough common understanding to give a reasonable enough estimate. This could include what is not included in the story, different possibilities of implementation, concerns, assumptions, or ideas.
+3. **Share their estimate (planning poker) as a t-shirt size** XS, S, M, L, or XL. XL is commonly referred as "a story that needs to be split before we can give an accurate enough estimate". XS is usually reserved for specifically small features.
```

Comment:
Does this mean that XL stories shouldn't go in the backlog until they have been split further? Otherwise it will skew the velocity calculation down the line.

Reply:
Yes, that is correct. It would happen naturally as they will be split during IPM or right before that.
However, it may be useful to keep an XL around it the project is unlikely to be implemented.